server/sonar-web/src/main/js/apps/organizations/store/reducer.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import uniq from 'lodash/uniq';
+import { actions } from './membersDuck';
+
+export const getOrganizationMembersLogins = (state, organization) => {
+  if (organization && state[organization]) {
+    return state[organization].members || [];
+  }
+  return [];
+};
+export const getOrganizationMembersState = (state, organization) =>
+  organization && state[organization] ? state[organization] : {};
+
+const organizationMembers = (state = {}, action = {}) => {
```

Comment:
there is a way to make this reducer more readable:

1. change `organizationMembers` to handle a single organization:
```
case actions.RECEIVE_MEMBERS:
  return {
     ...state,
    members: action.members.map(member => member.login)
  };
```
2. create another reducer `organizationsMembers`, that will do:
```
const organization = state[action.organization] || {};
return { ...state, [action.organization]: organizationMember(organization, action) };
```